ui/admin/tests/acceptance/host-catalogs/hosts/read-test.js

```diff
@@ -62,33 +62,55 @@ module('Acceptance | host-catalogs | hosts | read', function (hooks) {
     urls.hosts = `${urls.hostCatalog}/hosts`;
     urls.host = `${urls.hosts}/${instances.host.id}`;
     urls.unknownHost = `${urls.hosts}/foo`;
-    urls.newHost = `${urls.hosts}/new`;
-    // Generate resource couner
+
     authenticateSession({});
+    await visit(urls.hostCatalog);
   });
 
   test('visiting hosts', async function (assert) {
     assert.expect(2);
-    await visit(urls.hosts);
+    await click(`[href="${urls.hosts}"]`);
     await a11yAudit();
     assert.strictEqual(currentURL(), urls.hosts);
-    await visit(urls.host);
+
+    await click(`[href="${urls.host}"]`);
     await a11yAudit();
+
     assert.strictEqual(currentURL(), urls.host);
   });
 
-  test('cannot navigate to a role form without proper authorization', async function (assert) {
+  test('cannot navigate to a host form without proper authorization', async function (assert) {
     assert.expect(1);
     instances.host.authorized_actions =
       instances.host.authorized_actions.filter((item) => item !== 'read');
-    await visit(urls.hosts);
-    assert.notOk(find('main tbody .rose-table-header-cell:nth-child(1) a'));
+
+    await click(`[href="${urls.hosts}"]`);
+
+    assert.dom(`[href="${urls.host}"]`).doesNotExist();
   });
 
   test('visiting an unknown host displays 404 message', async function (assert) {
-    assert.expect(1);
+    assert.expect(2);
+    await click(`[href="${urls.hosts}"]`);
+    assert.dom(`[href="${urls.unknownHost}"]`).doesNotExist();
+
     await visit(urls.unknownHost);
     await a11yAudit();
-    assert.ok(find('.rose-message-subtitle').textContent.trim(), 'Error 404');
+
+    assert.strictEqual(
+      find('.rose-message-subtitle').textContent.trim(),
+      'Error 404'
+    );
```

Comment:
Also think this could've been:
```suggestion
    assert.dom('.rose-message-subtitle').hasText('Error 404');
```